### ISSUE TITLE
Regrid+Transform to logit doesn't work on MixMod

### ISSUE THREAD

**Issue by mamscience** [rvlenth/emmeans#378]
Hello,

Thanks for authoring and maintaining this excellent package. Firstly, it may be a bit unfair to speak of a bug as the underlying model is not supported, however, with all the extra tools provided one should be able to have a workaround. 

## Describe the bug
I've built a GLMMadaptive MixMod [where family= binomial/link="logit"], and the model appears to be performing well.
Emmeans doesn't recognize that it is on the logit scale.  This is seen when one contrasts the outcome: it can only calculate the absolute difference in logits. This is undesirable of course and odds-ratio or perhaps probabilities would be acceptable.

I discovered how to transform the emmGrid, but I can't get emmeans to regrid to a logit scale. It returns a emmGrid object that shows **nonEst**  for all parameters including the response variable. Of course, one can manually calculate it back with e.g. inv.logit, but this doesn't when one wants to cotrast. When contrasting, it also shows '**nonEst**'

## To reproduce - from https://drizopoulos.github.io/GLMMadaptive/articles/Multiple_Comparisons.html

```
set.seed(1234)
n <- 300 # number of subjects
K <- 4 # number of measurements per subject
t_max <- 15 # maximum follow-up time

# we construct a data frame with the design: 
# everyone has a baseline measurement, and then measurements at K time points
DF <- data.frame(id = rep(seq_len(n), each = K),
                 time = gl(K, 1, n*K, labels = paste0("Time", 1:K)),
                 sex = rep(gl(2, n/2, labels = c("male", "female")), each = K))

# design matrices for the fixed and random effects
X <- model.matrix(~ sex * time, data = DF)
Z <- model.matrix(~ 1, data = DF)

betas <- c(-2.13, 1, rep(c(1.2, -1.2), K-1)) # fixed effects coefficients
D11 <- 1 # variance of random intercepts

# we simulate random effects
b <- cbind(rnorm(n, sd = sqrt(D11)))
# linear predictor
eta_y <- as.vector(X %*% betas + rowSums(Z * b[DF$id, ]))
# we simulate binary longitudinal data
DF$y <- rbinom(n * K, 1, plogis(eta_y))
```

**Model, emmeans, regrid, contrast**
```
gm <- mixed_model(fixed = y ~ sex * time, random = ~ 1 | id, data = DF, family = binomial())
gm.emm<- emmeans(gm,specs =pairwise ~ sex | time)
regrid(gm.emm, transform="logit")
pairs(regrid(gm.emm, transform="logit"))
```

## Expected behavior
To transform variables to logit-scale

## Error received:

```
I bet you wanted to call this with just object[[1]] - use '[[]]' or which' if I'm wrong.
See '? emm_list' for more information
Invalid response predictions are flagged as non-estimable
time = Time1:
 sex    emmean SE df asymp.LCL asymp.UCL
 male   nonEst NA NA        NA        NA
 female nonEst NA NA        NA        NA
etc

```

Hope you can shed some light on this. Kind regards

**Comment by rvlenth** [comment]
The reason you are experiencing problems is that `regrid()` is doing the exact opposite of what you need! It is assuming everything is on the response scale (in this case the probability scale) and moving it to the logit scale. But in your case, the model was fitted on the logit scale, and all that is missing is the fact that the link function was not detected.  Instead, you should do:
```r
> gm_mc <- emmeans(gm, ~ sex | time, tran = "logit")
> ### (or   gm_mc <- update( emmeans(gm, ~ sex | time), tran = "logit")
> gm_mc
time = Time1:
 sex    emmean    SE  df asymp.LCL asymp.UCL
 male    -2.74 0.342 Inf     -3.41    -2.067
 female  -1.21 0.255 Inf     -1.71    -0.707

time = Time2:
 ...(several lines omitted) ...

Results are given on the logit (not the response) scale. 
Confidence level used: 0.95 
```
Note from the annotations that the estimates shown *are on the logit scale.* With `regrid()`, you were telling it that your estimated *probabilities* are -2.74, -1.21, etc., -- which are not valid values for probabilities -- and you want them on the logit scale. That's why you get nothing but NAs: they are all illegal.

Here are the estimated probabilities, based on simply updating to inform the software that the model was fitted on the logit scale.
```r
> confint(gm_mc, type = "response")
time = Time1:
 sex    response      SE  df asymp.LCL asymp.UCL
 male     0.0608 0.01954 Inf   0.03204    0.1124
 female   0.2302 0.04518 Inf   0.15361    0.3302

time = Time2:
 sex    response      SE  df asymp.LCL asymp.UCL
 male     0.1578 0.03660 Inf   0.09842    0.2432
 female   0.1923 0.04084 Inf   0.12451    0.2850

time = Time3:
 sex    response      SE  df asymp.LCL asymp.UCL
 male     0.0121 0.00653 Inf   0.00419    0.0345
 female   0.1997 0.04174 Inf   0.13010    0.2939

time = Time4:
 sex    response      SE  df asymp.LCL asymp.UCL
 male     0.2080 0.04300 Inf   0.13604    0.3046
 female   0.2224 0.04433 Inf   0.14750    0.3210

Confidence level used: 0.95 
Intervals are back-transformed from the logit scale 
```
Things are working pretty well now.

While we're at it, let's go on and try some pairwise comparisons
```r
> pairs(gm_mc, type = "response")
time = Time1:
 contrast      odds.ratio     SE  df null z.ratio p.value
 male / female     0.2164 0.0887 Inf    1  -3.735  0.0002

time = Time2:
 contrast      odds.ratio     SE  df null z.ratio p.value
 male / female     0.7867 0.2882 Inf    1  -0.655  0.5125

time = Time3:
 contrast      odds.ratio     SE  df null z.ratio p.value
 male / female     0.0492 0.0288 Inf    1  -5.145  <.0001

time = Time4:
 contrast      odds.ratio     SE  df null z.ratio p.value
 male / female     0.9184 0.3261 Inf    1  -0.240  0.8106

Tests are performed on the log odds ratio scale 
```
If you actually wanted differences instead of odds ratios, *that's* when `regrid()` is handy...
```r
> pairs(regrid(gm_mc), type = "response")
time = Time1:
 contrast      estimate     SE  df z.ratio p.value
 male - female  -0.1694 0.0477 Inf  -3.549  0.0004

time = Time2:
 contrast      estimate     SE  df z.ratio p.value
 male - female  -0.0346 0.0528 Inf  -0.655  0.5127

time = Time3:
 contrast      estimate     SE  df z.ratio p.value
 male - female  -0.1876 0.0417 Inf  -4.496  <.0001

time = Time4:
 contrast      estimate     SE  df z.ratio p.value
 male - female  -0.0144 0.0599 Inf  -0.240  0.8106
```
In the above, `regrid()` has re-parameterized the object so the estimates are on the response (probability) scale, and forget all about the logits, so it computes differences rather than odds ratios. If you were to do `regrid(regrid(gm_mc), transform = "logit")`, you will get something equivalent to `gm_mc` (but internally different).

**Comment by mamscience** [comment]
Amazing, thank you very much! Exactly what was needed to solve!!
Thank you for your support.
